Blish HUD/GameServices/ArcDpsService.cs

```diff
@@ -101,47 +90,56 @@ private void DispatchSkillSubscriptions(object sender, RawCombatEventArgs eventH
         /// </summary>
         public event EventHandler<RawCombatEventArgs> RawCombatEvent;
 
-        protected override void Initialize()
-        {
-            Common = new CommonFields();
-            _stopwatch = new Stopwatch();
-            _server = new SocketListener(10, 200_000);
+        protected override void Initialize() {
+            this.Common             =  new CommonFields();
+            _stopwatch              =  new Stopwatch();
+            _server                 =  new SocketListener(200_000);
             _server.ReceivedMessage += MessageHandler;
-#if DEBUG
-            RawCombatEvent += (a, b) => { Interlocked.Increment(ref Counter); };
-#endif
+            #if DEBUG
+            this.RawCombatEvent += (a, b) => { Interlocked.Increment(ref Counter); };
+            #endif
         }
 
-        protected override void Load()
-        {
-            _server.Start(new IPEndPoint(IPAddress.Loopback, 8214));
+        protected override void Load() {
+            Gw2Mumble.Info.ProcessIdChanged += Start;
             _stopwatch.Start();
         }
 
-        protected override void Unload()
-        {
+        private void Start(object sender, ValueEventArgs<uint> value) {
+            if (this.Loaded) _server.Start(new IPEndPoint(IPAddress.Loopback, GetPort(value.Value)));
```

Comment:
Do we stop listening if there was already a listener going?

For instance, if Blish HUD is left open between sessions and has multiple PIDs in the lifetime, will it properly stop the first listener?

Reply:
For now, the gc will handle it. Between the runs of gw2 the socket would just wait for something to happen, which will never happen. When gw2 then starts again, the new socket will be initialized and the old one will be taken care of by gc.
I could add some handling here: https://github.com/blish-hud/Blish-HUD/blob/e7c443ea6703c7ee23befbb2637fbea089473c0f/Blish%20HUD/GameServices/ArcDps/SocketListener/SocketListener.cs#L59-L68
There is a disconnect enum variant that could either stop the listener altogether or try to reconnect a few times, etc.

Reply:
That makes sense.  Could maybe kill the listener when we detect the game closing (event in GameIntegration service), but not a big deal.

Thanks for clarifying!